arviz/stats/stats.py

```diff
@@ -1126,3 +1136,232 @@ def waic(data, pointwise=False, scale="deviance"):
                 "waic_scale",
             ],
         )
+
+
+def loo_pit(y, y_hat, log_weights):
```

Comment:
Could we have a wrapper for inference data, where y and yhat could be strings? Or is there one already?

Reply:
There is no wrapper, but the conversion from idata+strings is already done in the plot, so making the wrapper is no problem at all. Once it is done though, should this function and the wrapper be public? Should they be the same function?

Reply:
You could follow ess / rhat format? So main function would be public.

Reply:
I am not sure on how to export their format here. ess and rhat have the basic ufunc, which is private and the public function. And the public function `data` argument accepts either inference data or array. The basic ufunc is already present, and to follow the ess/rhat behaviour, the public function should accept both `loo_pit(idata: InferenceData, y: str, y_hat: str)` and `loo_pit(y: array-like, y_hat: array-like, log_weights: array-like)`, which do not seem very compatible

Reply:
I think that format is ok

    idata, y: str, yhat: str
    y: ndarray, yhat: ndarray

Do you think there would be cases where 1) user calculates `yhat` after sampling -->

    idata
    y : str
    yhat : ndarray

or 2) where user has forgot to include data to idata -->

    idata
    y: ndarray
    yhat: str

And are we assuming that yhat is in posterior_predictive and y in observed_data. Maybe there could be kwarg that can change these defaults.

I know that is a bit complicated, but atleast I don't always remember to do posterior_predictive, and some interfaces don't have observed_data by default. So having flexible interface is great.

Reply:
I have modified the call signature to `loo_pit(idata, y, y_hat, log_weights)` and the same in `plot_loo_pit()` (arguments are passed directly to `loo_pit`). Now, the following possibilities are accepted:

1. `loo_pit(idata: InferenceData, y: str)`
1. `loo_pit(idata: InferenceData, y: str, y_hat: str)`
1. `loo_pit(idata: InferenceData, y: str, y_hat: array-like)`
1. `loo_pit(idata: InferenceData, y: array-like, y_hat: str)`
1. `loo_pit(idata: InferenceData, y: array-like, y_hat: array-like)`
1. `loo_pit(idata: InferenceData, y: str, y_hat: str, log_weights: array-like)`
1. `loo_pit(idata: InferenceData, y: str, y_hat: array-like, log_weights: array-like)`
1. `loo_pit(idata: InferenceData, y: array-like, y_hat: str, log_weights: array-like)`
1. `loo_pit(y: array-like, y_hat: array-like, log_weights: array-like)`

With the array inputs taking priority over strings or None, that is, if `log_weights` is array-like, these values are used instead of checking for a `log_likelihood` variable in the InferenceData. For now, they are all optional _positional_ arguments, which is not ideal. I would set _all_ 4 of them to be optional _named_ arguments, what do you think? Leaving `idata` as positional and set the other three to named arguments also makes sense.